docs/resource/README.md

```diff
@@ -101,10 +101,13 @@ as specified in the [Resource SDK specification](https://github.com/open-telemet
 |---|---|---|---|---|
 | `service.instance.id` | string | The string ID of the service instance. [1] | `my-k8s-pod-deployment-1`; `627cc493-f310-47de-96bd-71410b7dec09` | Recommended |
 | `service.namespace` | string | A namespace for `service.name`. [2] | `Shop` | Recommended |
+| `service.type` | string | The service type in reverse domain name notation. [3] | `io.opentelemetry.collector`; `io.redis`; `org.apache.kafka` | Recommended |
```

Comment:
Would be good to eventually have a page / section dedicated to describing the known taxonomy of `service.type` values so that consumers can key off those values for different behavior. Without listing the known values, the usefulness of `service.type` is diminished since the names will only be relevant to whoever sets the names.

Reply:
>  Or do you mean just Otel's products, like Collector?

The collector seems particularly important.

> Do you mean a page here in this repo that lists every possible software that we are aware of?

The idea in my head isn't to try to catalog all the possible systems up front, but to allow folks to define new values as needed. One area I think `service.type` would be particularly useful is for the various collector receivers which scrape / poll / query other systems and convert to the OpenTelemetry data model. For example, it would be very useful to be able to assign something like `service.type=io.apache.kafka.broker` to the data from the [kafkametricsreceiver](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/kafkametricsreceiver), or `service.type=io.apache.flink.worker` to the data from [flinkmetricsreceiver](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/flinkmetricsreceiver). I don't think `service.type` is applicable for all receivers since some  like the [mysqlreceiver](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/mysqlreceiver) are all about pulling / translating generic data from a db. But for the receivers which specifically pull telemetry for a particular _type_ of system, it would be very useful to be able to label the data as as associated with a `service.type`. (Side note, it would also be use for these same receivers to have a configurable `service.name` configuration property do define the logical name / role of the observed system. In my experience, most just send the data along with no `service.name`, which has confused me in the past since it doesn't appear to conform to resource semantic conventions.)

Reply:
Good point regarding Collector setting the right `service.name`, `service.type` in product-specific receivers. 

Perhaps a good place to have that list is in Collector's contrib repo. I think receiver component's [metadata](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/receiver/mysqlreceiver/metadata.yaml) already allows recording resource attributes, probably this is where the `server.type` should go as well (and we can then autogenerate the list of all known service.type values handled by Collector). @open-telemetry/collector-maintainers can you please help clarify this? Do resource_attributes in metadata.yaml control what Resource is emitted by the particular receiver?

Reply:
>Do resource_attributes in metadata.yaml control what Resource is emitted by the particular receiver?

The `resource_attributes` key controls which resource attributes are available for a scraping metrics receiver to emit. At a minimum, we can always set a single `service.type` in scraping receivers and document this in their readmes. If we wanted to assemble these into a list we would likely want to add support setting static values for resource attributes inside metadata.yaml.

Reply:
I do think there is real value in having a list of "known" service.type(s). This would make more clear the relationship between a service.type and the expected data collected for said service. Having this would allow users to understand what metrics and attributes they can use for a product they deploy. ex:
`service.type=otc` -> list of metrics (otelcol_receiver..., otelcol_processor..., etc.)

Reply:
+1 for seeing a lot of value in this list. End-users will not be clear about the `service.type` initially, so if we give them something more tangible to look this up, it's easier to figure out what they might want to set themselves.

If I understand this correctly, this can also be used for things like Wordpress (org.wordpress?), WooCommerce (com.woo?), etc or any other prepackaged software?

Reply:
>I do think there is real value in having a list of "known" service.type(s).

@jaronoff97 can you clarify what do you mean by "known" list? This can't be an exhaustive list, we can't list every product in the world here, so what exactly do you expect to see?

Reply:
>If I understand this correctly, this can also be used for things like Wordpress (org.wordpress?), WooCommerce (com.woo?), etc or any other prepackaged software?

@svrnm yes, that is correct.

I am not sure I understand what you expect when you say "a lot of value in this list". I have 3 examples there. Would adding a few more help?

Reply:
Added one more example.

Reply:
> @svrnm yes, that is correct.

Thanks. 
 
> I am not sure I understand what you expect when you say "a lot of value in this list". I have 3 examples there. Would adding a few more help?

I was referring to what @jaronoff97's comment "I do think there is real value in having a list of "known" service.type(s).". For the current PR the number of examples is enough. I also will make a note that we need to build out a section about `service` in the documentation, which eventually can provide some more details on that.

Reply:
@tigrannajaryan i think the examples you have now are great and this change can be merged in. I was referring to having a separate list of known `service.types` similar to @jack-berg 's suggestion above:
> For example, it would be very useful to be able to assign something like service.type=io.apache.kafka.broker to the data from the [kafkametricsreceiver](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/kafkametricsreceiver), or service.type=io.apache.flink.worker to the data from [flinkmetricsreceiver](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/flinkmetricsreceiver). I don't think service.type is applicable for all receivers since some like the [mysqlreceiver](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/mysqlreceiver) are all about pulling / translating generic data from a db. But for the receivers which specifically pull telemetry for a particular type of system, it would be very useful to be able to label the data as as associated with a service.type

I'm looking for something to enhance the metadata.yaml files that collector receivers have right now in order to make assertions on expected data emitted if i observe a specific `service.type`. This list would essentially just be a mapping from receiver name -> service.type. I'm not sure if this list should exist here or in the collector. Regardless, this is a follow up.

Reply:
@jaronoff97 I think that list/mapping belongs in the Collector. Collector maintainers will need to decide how they want to do it, after this PR is merged. I agree that we want receivers to include a relevant `service.type`.

Reply:
Resolving, looks like we are in agreement.